### PR TITLE
Implement sidebar pages

### PR DESCRIPTION
**Pull request by mohamdlog**
This PR refactors the routing structure and adds a `ProfileLayout` component for better organization of profile-related pages.

**Key Changes:**  

`App.js:`
- Wrapped all profile routes under the new ProfileLayout for cleaner routing.
- Removed old profile routes from the main routing structure.

`ProfileLayout.js:`
- Created a layout with a sidebar for profile navigation and an Outlet for nested routes.

`Sidebar:`
- Updated links to support nested profile routes like My Listings, List Item, Conversations, and Delete Account.

**Minor Changes:**
- Set up sidebar pages and files for development.

### CODE REVIEW

**Review by shagunsharmaslu — APPROVED** [review]
Looks good to me.

**Comment by vishalreddy9866** [comment]
Looks good. The addition of the sidebar for navigation is a nice touch.

**Review by vishalreddy9866 — APPROVED** [review]

File: frontend/src/pages/profile/profilePage/index.jsx [hunk 1]
```diff
@@ -55,4 +53,4 @@ const ProfilePage = () => {
     );
 };
 
-export default ProfilePage;
\ No newline at end of file
+export default ProfilePage;
```
Comment (nlp-saiteja) [inline frontend/src/pages/profile/profilePage/index.jsx:56]: The Code Structure looks clean and functions as intended

**Review by nlp-saiteja — APPROVED** [review]
The Code Structure looks clean and functions as intended

**Pull request merged**